theano/compile/tests/test_function_module.py

```diff
@@ -246,27 +251,25 @@ def test_copy_share_memory(self):
         # SharedVariable for tests, one of them has update
         y = theano.shared(value=1)
         z = theano.shared(value=2)
-        out = T.tanh((x+y+2)/(x+z-0.2)**2)
+        out = T.tanh((x + y + 2) / (x + z - 0.2)**2)
 
         # Test for different linkers
-        for mode in ["FAST_RUN","FAST_COMPILE"]:
-            ori = theano.function([x], [out], mode=mode,updates={z:z+1})
+        for mode in ["FAST_RUN", "FAST_COMPILE"]:
+            ori = theano.function([x], [out], mode=mode, updates={z: z + 1})
             cpy = ori.copy(share_memory=True)
 
             # Test if memories shared
             storage_map_ori = ori.fn.storage_map
             storage_map_cpy = cpy.fn.storage_map
-            fgraph_ori = ori.maker.fgraph
             fgraph_cpy = cpy.maker.fgraph
 
             # Assert intermediate and Constants storages are shared.
             # and output stoarges are not shared
             i_o_variables = fgraph_cpy.inputs + fgraph_cpy.outputs
             ori_storages = storage_map_ori.values()
-            for key in storage_map_cpy.keys():
-                storage = storage_map_cpy[key]
-                if key not in i_o_variables or isinstance(key, theano.tensor.Constant):
-                    self.assertTrue(any([ storage is s for s in ori_storages]))
+            l = [val for key, val in storage_map_cpy.items()
+                 if key not in i_o_variables or isinstance(key, theano.tensor.Constant)]
+            self.assertTrue(any([val is s for val in l for s in ori_storages]))
```

Comment:
I proposed the change because it is usually faster and to me is easier to read! :) 
I guess it's a matter of taste.

You are right that the two codes are not doing the same thing, the `any` should have been in the list comprehension. To fix this and address your readability concerns too, we can change it to:
```python
l = [val for key, val in storage_map_cpy.items()
      if key not in i_o_variables or isinstance(key, theano.tensor.Constant)]
for storage in l:
    self.assertTrue(any([storage is s for s in ori_storages]))
```